spectator/http_client.cc

```diff
@@ -93,10 +88,10 @@ class CurlHandle {
     curl_easy_setopt(handle_, CURLOPT_TIMEOUT_MS, millis);
   }
 
-  void post_payload(std::shared_ptr<char> payload, size_t size) {
-    payload_ = std::move(payload);
+  void post_payload(const void* payload, size_t size) {
```

Comment:
Why switch to `void*` here?

Reply:
Before we used a `shared_ptr` because we would store the payload in case we were to use async sends, now it's just a sink argument. When we return from the call we do not need the payload anymore.

Reply:
Also `void*` instead of `char*` because we could send any binary serialized structure, not just strings since we take a pointer and a size in bytes.